Ability to specify key for IdentityUser

FairlieAgile opened the pull request.

thomasduft commented:
Hello @FairlieAgile 

Finally after some time I quickly looked at the PR. Having also pulled it locally I see that some Identity-related Integration-Tests still fail 
![image](https://user-images.githubusercontent.com/11708066/166640073-622aeb2a-a155-4bc5-99e5-76d1bf3c102a.png).

Did you run them as well locally without seeing any failing tests? Using VSCode you can just hit `CTRL-P` and then type `task test`and hit ENTER. There is a test-task configured for VSCode 
![image](https://user-images.githubusercontent.com/11708066/166641180-7c0008f4-1029-4781-8f3f-cb63bdddcd26.png).

BTW I would appreciate if you could create a separate server project (for example ServerWithCustomKeys) within the samples directory so we could showcase running the openiddict-ui libs with defaults and custom keys. Your PR would also mean a breaking change regarding the registering openiddict-ui-identity APIs.

FairlieAgile commented:
Hi,

Thanks for the feedback. I have now corrected so the tests run properly. This highlighted an issue with RoleApiService which I have corrected.

I have also added in some tests for Identity and Roles with the Custom Key.

This change will not create a breaking change. I have removed the changes to the existing Server sample to demonstrate this. If the default Identity is used the constructor will treat this as a string.

Cheers
Paul

thomasduft commented:
Fixes #16
